src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1317,81 +1304,56 @@ std::unique_ptr<StorageTargetWHandle> SQLStorage::allocateTargetFile(bool from_d
 class SQLTargetRHandle : public StorageTargetRHandle {
  public:
   SQLTargetRHandle(const SQLStorage& storage, Uptane::Target target)
-      : db_path_(storage.dbPath()),
-        db_(db_path_),
-        target_(std::move(target)),
-        size_(0),
-        read_size_(0),
-        closed_(false),
-        blob_(nullptr) {
+      : db_path_(storage.dbPath()), db_(db_path_), target_(std::move(target)), size_(0) {
     StorageTargetRHandle::ReadError exc("could not read file " + target_.filename() + " from sql storage");
 
     auto exists = storage.checkTargetFile(target_);
     if (!exists) {
       throw exc;
     }
-    auto statement = db_.prepareStatement<std::string>("SELECT rowid FROM target_images_data WHERE filename = ?;",
-                                                       target_.filename());
 
-    if (statement.step() != SQLITE_ROW) {
-      LOG_ERROR << "Statement step failure: " << db_.errmsg();
-      throw std::runtime_error("Could not find target file");
-    }
-
-    row_id_ = statement.get_result_col_int(0);
     size_ = *exists;
     partial_ = size_ < target_.length();
-    if (sqlite3_blob_open(db_.get(), "main", "target_images_data", "image_data", row_id_, 0, &blob_) != SQLITE_OK) {
-      LOG_ERROR << "Could not open blob: " << db_.errmsg();
+    image_path_ = storage.images_path_ / target_.filename();
+    stream_.open(image_path_.string());
+    if (!stream_.good()) {
+      LOG_ERROR << "Could not open image: " << storage.images_path_ / target_.filename();
       throw exc;
     }
   }
 
-  ~SQLTargetRHandle() override {
-    if (!closed_) {
-      SQLTargetRHandle::rclose();
-    }
-  }
+  ~SQLTargetRHandle() override { SQLTargetRHandle::rclose(); }
 
   size_t rsize() const override { return size_; }
 
   size_t rread(uint8_t* buf, size_t size) override {
-    if (read_size_ + size > size_) {
-      size = size_ - read_size_;
-    }
-    if (size == 0) {
-      return 0;
-    }
-
-    if (sqlite3_blob_read(blob_, buf, static_cast<int>(size), static_cast<int>(read_size_)) != SQLITE_OK) {
-      LOG_ERROR << "Could not read from blob: " << db_.errmsg();
-      return 0;
+    auto red_bytes = stream_.readsome(reinterpret_cast<char*>(buf), static_cast<std::streamsize>(size));
+    if (!stream_.good()) {
```

Comment:
This check looks superfluous to me, `read()` will throw an exception if something went wrong.